src/pymor/reductors/ph/ph_irka.py

```diff
@@ -0,0 +1,89 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.models.iosys import PHLTIModel
+from pymor.reductors.h2 import GenericIRKAReductor, OneSidedIRKAReductor
+from pymor.reductors.ph.basic import PHLTIPGReductor
+
+
+class PHIRKAReductor(GenericIRKAReductor):
+    """PH-IRKA reductor.
+
+    Parameters
+    ----------
+    fom
+        The full-order |PHLTIModel| to reduce.
+    mu
+        |Parameter values|.
+    """
+
+    def __init__(self, fom, mu=None):
+        assert isinstance(fom, PHLTIModel)
+        super().__init__(fom, mu=mu)
+
+    def reduce(self, rom0_params, tol=1e-4, maxit=100, num_prev=1, projection='orth', conv_crit='sigma',
+               compute_errors=False):
+        r"""Reduce using pH-IRKA.
+
+        It uses IRKA as the intermediate reductor.
+
+        See :cite:`GPBV12`.
+
+        Parameters
+        ----------
+        rom0_params
+            Can be:
+
+            - order of the reduced model (a positive integer),
+            - dict with `'sigma'`, `'b'`, `'c'` as keys mapping to
+              initial interpolation points (a 1D |NumPy array|), right
+              tangential directions (|NumPy array| of shape
+              `(len(sigma), fom.dim_input)`), and left tangential directions
+              (|NumPy array| of shape `(len(sigma), fom.dim_input)`),
+            - initial reduced-order model (|PHLTIModel|).
```

Comment:
This should be tested.

Reply:
I think it would also be good to allow both `LTIModels` and `PHLTIModels` as initial ROMs.

Reply:
I'm a little bit confused, I think we don't allow `LTIModels` and the algorithm will break since `PHLTIPGReductor` can not handle `LTIModels`.